ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdCalculationService.java

```diff
@@ -81,6 +83,25 @@ public void setApplicationContext(@Nonnull ApplicationContext applicationContext
 		this.applicationContext = applicationContext;
 	}
 
+	PersistedInstanceIdStatisticsMBean collectStatistics(MirroredObject<?> mirroredObject) {
+		return jobStatistics.computeIfAbsent(mirroredObject.getCollectionName(), x -> new PersistedInstanceIdStatistics());
+	}
+
+	// Initializes readyForNumberOfPartitions for collections that are already calculated
+	void initializeStatistics() {
+		Set<Integer> numberOfPartitionsToCalculate = getNumberOfPartitionsToCalculate();
+		jobStatistics.replaceAll((collectionName, statistics) -> {
```

Comment:
This method `initializeStatistics` currently seems to rely on `collectStatistics` having been called before, so that `jobStatistics` is populated with the collection names. This class already seems to have the responsibility of determining "which collections should have an index?" (see `calculatePersistedInstanceId()`), so maybe this method could reuse that responsibility here, to reduce the requirement that some other method must be called before this `initialize`-method?

Reply:
Good suggestion! Removing the requirement of calling the methods in order here reduces the complexity a bit. 👍

Reply:
Nice. Simpler code and easier to follow.